test/e2e/checkpoint_test.go

```diff
@@ -1555,6 +1555,132 @@ var _ = Describe("Podman checkpoint", func() {
 		os.Remove(fileName)
 	})
 
+	It("podman checkpoint container with export and verify non-default runtime", func() {
+		SkipIfRemote("podman-remote does not support --runtime flag")
+		// This test triggers the edge case where:
+		// 1. Default runtime is crun
+		// 2. Container is created with runc
+		// 3. Checkpoint without setting --runtime into archive
+		// 4. Restore without setting --runtime from archive
+		// It should be expected that podman identifies runtime
+		// from the checkpoint archive.
+
+		// Prevent --runtime arg from being implicitly set to force using default
+		// runtime unless explicitly set through passed args.
+		podmanTest.PodmanMakeOptions = func(args []string, noEvents, noCache bool) []string {
+			defaultArgs := podmanTest.makeOptions(args, noEvents, noCache)
+			for i := range args {
+				// Runtime is set explicitly, so we should keep --runtime arg.
+				if args[i] == "--runtime" {
+					return defaultArgs
+				}
+			}
+			for i := range defaultArgs {
+				// Remove implicit --runtime arg
+				if defaultArgs[i] == "--runtime" {
+					return append(defaultArgs[:i], defaultArgs[i+2:]...)
+				}
+			}
+			return defaultArgs
+		}
+
+		cmd := exec.Command("runc", "--help")
+		if err := cmd.Start(); err != nil {
+			Skip("Test requires runc")
+		}
+		if err := cmd.Wait(); err != nil {
+			Skip("Test requires runc")
+		}
+		cmd = exec.Command("crun", "--help")
+		if err := cmd.Start(); err != nil {
+			Skip("Test requires crun")
+		}
+		if err := cmd.Wait(); err != nil {
+			Skip("Test requires crun")
+		}
+
+		session := podmanTest.Podman([]string{
+			"info",
+			"--format",
+			"{{.Host.OCIRuntime.Name}}",
+		})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+		defaultRuntime := session.OutputToString()
+		if !strings.Contains(defaultRuntime, "crun") {
```

Comment:
Okay, why `Contains`? You're asking for `.Name`. It should be safe to use `Equals` here.